src/charm.py

```diff
@@ -584,6 +585,18 @@ def _update_cert(self):
         self.container.exec(["update-ca-certificates", "--fresh"]).wait()
         subprocess.run(["update-ca-certificates", "--fresh"])
 
+    def _check_disk_space(self):
+        try:
+            #database_storage = self.model.storages.get('database', []) #self.model.storages["database"]
+            database_storage = self.model.storages["database"]
+            if database_storage and shutil.disk_usage(database_storage[0].location).free < 1024**3: # type: ignore
```

Comment:
```suggestion
            
            if database_storage:
                # NOTE: we measure the disk space from the charm container because it shares the same storage as the workload container
                free_disk_space = shutil.disk_usage(database_storage[0].location).free
                
             if free_disk_space < 1024**3:
```

`type: ignore` is a leftover from a previous approach!